### ISSUE TITLE
Question about the anchor generator

### ISSUE THREAD

**Issue by d5423197** [meituan/YOLOv6#617]
### Before Asking

- [X] I have read the [README](https://github.com/meituan/YOLOv6/blob/main/README.md) carefully. 我已经仔细阅读了README上的操作指引。

- [X] I want to train my custom dataset, and I have read the [tutorials for training your custom data](https://github.com/meituan/YOLOv6/blob/main/docs/Train_custom_data.md) carefully and organize my dataset correctly; (FYI: We recommand you to apply the config files of xx_finetune.py.) 我想训练自定义数据集，我已经仔细阅读了训练自定义数据的教程，以及按照正确的目录结构存放数据集。（FYI: 我们推荐使用xx_finetune.py等配置文件训练自定义数据集。）

- [X] I have pulled the latest code of main branch to run again and the problem still existed. 我已经拉取了主分支上最新的代码，重新运行之后，问题仍不能解决。

### Search before asking

- [X] I have searched the YOLOv6 [issues](https://github.com/meituan/YOLOv6/issues) and found no similar questions.

### Question

Hello there,

This is my second time reading Yolov6 code. It is quite different from the one few months ago. What does cell_half_size mean?
Actually I dont actually understand what generate_anchors do.

Thanks

**Comment by Chilicyy** [comment]
`generate_anchors` is used for ATSS label assignment to generate some anchors. `cell_half_size` is a hyperameter for generating anchors from a anchor point in the center of the grid.

**Comment by d5423197** [comment]
So cell_half_size  is like a fixed anchor box size?

`anchor = torch.stack(
                [
                    shift_x - cell_half_size, shift_y - cell_half_size,
                    shift_x + cell_half_size, shift_y + cell_half_size
                ],`
I know shift_x and shift_y is the center for the grid cell. So from this code, I am guessing cell_half_size is half of the anchor width and length? The anchor is a square?

**Comment by Chilicyy** [comment]
Yes,  the generated anchor is a square.

**Comment by d5423197** [comment]
Isn't YOLOV6 anchor free?

**Comment by Chilicyy** [comment]
YOLOv6 is an anchor-free detection, which uses TAL label assignment with ATSS warmup strategy. The generated anchors are just used in ATSS for the first 4 epoches.

**Comment by d5423197** [comment]
Thanks. I wanna explore more about the intuition behind this. I mean in my opinion, the anchor size is too big.

**Comment by Chilicyy** [comment]
This size is a emprical value.
